### PR TITLE
Update upgrade guide

### LINKED ISSUE #5766: Issue about the API of NCCL
Hi,

It seems that the API of NCCL described in the document, like `cupy.cuda.nccl`, is not working in the latest Cupy. It will throw an import error saying `cupy.cuda `does not have `nccl`. After checking the source code, I found a workaround is to `import cupy_backends `, then change all the `cupy.cuda.nccl...` to `cupy_backends.cuda.libs.nccl`. Is this expected? or I misused something?

### PR DESCRIPTION
**Pull request by kmaehashi**
Closes #5766
Refs #5818 (keeping this issue open as this PR is not covering all changes in v10)

Auto-backport to v9 should fail, I'll take care of it.

### CODE REVIEW

File: docs/source/upgrade.rst [hunk 1]
```diff
@@ -7,17 +7,22 @@ This is a list of changes introduced in each release that users should be aware
...
 
-Dropping CUDA 9.2 Support
-----------------------------
+Dropping CUDA 9.2 / 10.0 / 10.1 Support
+---------------------------------------
 
-CUDA 9.2 is no longer supported.
-Use CUDA 10.0 or later.
+CUDA 10.1 or earlier is no longer supported.
+Use CUDA 9.2 or later.
```
Comment (emcastillo) [inline docs/source/upgrade.rst:14]:
```suggestion
Use CUDA 10.2 or later.
```

File: docs/source/upgrade.rst [hunk 2]
```diff
@@ -106,6 +111,17 @@ NCCL and cuDNN No Longer Included in Wheels
...
+cuTENSOR Enabled in Wheels
+--------------------------
+
+cuTENSOR can now be used when installing CuPy via wheels.
+
+``cupy.cuda.{nccl,cudnn}`` Modules Needs Explicit Import
+--------------------------------------------------------
+
+Previously ``import cupy.cuda`` automatically made ``cupy.cuda.nccl`` and ``cupy.cuda.cudnn`` modules available.
+In CuPy v9, you need to import them explicitly to use these modules (i.e., ``import cupy.cuda.nccl`` / ``import cupy.cuda.cudnn``.)
```
Comment (emcastillo) [inline docs/source/upgrade.rst:123]:
```suggestion
In CuPy v9, these modules need to be explicitly imported (i.e., ``import cupy.cuda.nccl`` / ``import cupy.cuda.cudnn``.)
```

File: docs/source/upgrade.rst [hunk 3]
```diff
@@ -106,6 +111,17 @@ NCCL and cuDNN No Longer Included in Wheels
...
 
+cuTENSOR Enabled in Wheels
+--------------------------
+
+cuTENSOR can now be used when installing CuPy via wheels.
+
+``cupy.cuda.{nccl,cudnn}`` Modules Needs Explicit Import
+--------------------------------------------------------
+
+Previously ``import cupy.cuda`` automatically made ``cupy.cuda.nccl`` and ``cupy.cuda.cudnn`` modules available.
```
Comment (emcastillo) [inline docs/source/upgrade.rst:122]:
```suggestion
Previously ``cupy.cuda.nccl`` and ``cupy.cuda.cudnn`` modules were available when directly importing CuPy with ``import cupy.cuda``
```

Comment (kmaehashi) [reply docs/source/upgrade.rst:122]: To be more precise, just doing `import cupy` made these modules available.
```suggestion
Previously ``cupy.cuda.nccl`` and ``cupy.cuda.cudnn`` modules were automatically imported.
```

Comment (emcastillo) [reply docs/source/upgrade.rst]: yeah, this sounds much better :)

**Review by emcastillo — APPROVED** [review]

**Comment by emcastillo** [comment]
Jenkins, test this please

**Pull request merged**